tests/model_registry/model_catalog/utils.py

```
@@ -275,3 +282,122 @@ def assert_source_error_state_message(
+def execute_get_call(
+    url: str, headers: dict[str, str], verify: bool | str = False, params: dict[str, Any] | None = None
+) -> requests.Response:
+    LOGGER.info(f"Executing get call: {url}")
+    if params:
+        LOGGER.info(f"params: {params}")
+    resp = requests.get(url=url, headers=headers, verify=verify, timeout=60, params=params)
+    LOGGER.info(f"Encoded url from requests library: {resp.url}")
+    if resp.status_code not in [200, 201]:
+        # Raise custom exception for 401 errors that can be retried (OAuth/kube-rbac-proxy initialization)
+        if resp.status_code == 401:
+            raise TransientUnauthorizedError(f"Get call failed for resource: {url}, 401: {resp.text}")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)